modules/lang-python/src/main/java/org/opensearch/python/PythonFieldScript.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.python;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.index.LeafReaderContext;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.SandboxPolicy;
+import org.graalvm.polyglot.Value;
+import org.opensearch.script.FieldScript;
+import org.opensearch.search.lookup.SearchLookup;
+
+public class PythonFieldScript {
+    private static final Logger logger = LogManager.getLogger();
+
+    public static FieldScript.Factory newFieldScriptFactory(String code) {
+        return new FieldScript.Factory() {
+            @Override
+            public boolean isResultDeterministic() {
+                return true;
+            }
+
+            @Override
+            public FieldScript.LeafFactory newFactory(
+                    Map<String, Object> params, SearchLookup lookup) {
+                return newFieldScript(code, params, lookup);
+            }
+        };
+    }
+
+    private static FieldScript.LeafFactory newFieldScript(
+            String code, Map<String, Object> params, SearchLookup lookup) {
+        logger.info("Executing python code: {}", code);
```

Comment:
Why not combining these 3 lines log into one line? And I think it should be debug level?